File: DataSources/PostgreSQL/server/spring-boot-jersey/src/main/java/com/server/reveal/DataSourceProvider.java
package com.server.reveal;

import com.infragistics.reveal.sdk.api.IRVDataSourceProvider;
import com.infragistics.reveal.sdk.api.IRVUserContext;
import com.infragistics.reveal.sdk.api.model.*;

public class DataSourceProvider implements IRVDataSourceProvider {
    public RVDataSourceItem changeDataSourceItem(IRVUserContext userContext, String dashboardsID, RVDataSourceItem dataSourceItem) {

        if (dataSourceItem instanceof RVPostgresDataSourceItem postgresDataSourceItem) {

            //update underlying data source
            changeDataSource(userContext, dataSourceItem.getDataSource());

            //only change the table if we have selected our custom data source item
            if (dataSourceItem.getId() == "MyPostgresDataSourceItem") {
                postgresDataSourceItem.setTable("orders");
            }
        }
        return dataSourceItem;
    }

    public RVDashboardDataSource changeDataSource(IRVUserContext userContext, RVDashboardDataSource dataSource) {

        if (dataSource instanceof RVPostgresDataSource postgresDataSource) {
            postgresDataSource.setHost("localhost");
            postgresDataSource.setDatabase("database");
            postgresDataSource.setSchema("public");
        }
        return dataSource;
    }
}

File: DataSources/PostgreSQL/server/aspnet/RevealSdk.Server/Reveal/DataSourceProvider.cs
using Reveal.Sdk;
using Reveal.Sdk.Data;
using Reveal.Sdk.Data.PostgreSQL;

namespace RevealSdk.Server.Reveal
{
    public class DataSourceProvider : IRVDataSourceProvider
    {
        public Task<RVDataSourceItem> ChangeDataSourceItemAsync(IRVUserContext userContext, string dashboardId,
            RVDataSourceItem dataSourceItem)
        {
            if (dataSourceItem is RVPostgresDataSourceItem postgresDataSourceItem)
            {
                //update underlying data source
                ChangeDataSourceAsync(userContext, postgresDataSourceItem.DataSource);

                //only change the table if we have selected our custom data source item
                if (postgresDataSourceItem.Id == "MyPostgresDataSourceItem")
                {
                    postgresDataSourceItem.Table = "orders";
                }
            }

            return Task.FromResult(dataSourceItem);
        }

        public Task<RVDashboardDataSource> ChangeDataSourceAsync(IRVUserContext userContext,
            RVDashboardDataSource dataSource)
        {
            if (dataSource is RVPostgresDataSource postgresDataSource)
            {
                postgresDataSource.Host = "localhost";
                postgresDataSource.Database = "database";
                postgresDataSource.Schema = "public";
            }

            return Task.FromResult(dataSource);
        }
    }
}